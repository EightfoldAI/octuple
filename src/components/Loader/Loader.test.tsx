import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { Loader, LoaderProps, LoaderSize } from './';
import { render } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

const loaderProps: LoaderProps = {
  classNames: 'my-loader-class',
  'data-testid': 'test-loader',
};

describe('Loader', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders without crashing', () => {
    const { container, getByTestId } = render(<Loader {...loaderProps} />);
    const loader = getByTestId('test-loader');
    expect(() => container).not.toThrowError();
    expect(loader).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Loader accepts custom classes', () => {
    const { container } = render(<Loader {...loaderProps} />);
    expect(container.querySelector('.my-loader-class')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Loader is large', () => {
    const { container } = render(
      <Loader {...loaderProps} size={LoaderSize.Large} />
    );
    expect(container.querySelector('.large')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Loader is medium', () => {
    const { container } = render(
      <Loader {...loaderProps} size={LoaderSize.Medium} />
    );
    expect(container.querySelector('.medium')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Loader is small', () => {
    const { container } = render(
      <Loader {...loaderProps} size={LoaderSize.Small} />
    );
    expect(container.querySelector('.small')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
