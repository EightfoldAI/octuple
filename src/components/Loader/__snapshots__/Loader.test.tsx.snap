// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Loader accepts custom classes 1`] = `
<div>
  <div
    class="loader-container my-loader-class"
    data-testid="test-loader"
  >
    <div
      class="dot small"
    />
    <div
      class="dot small"
    />
    <div
      class="dot small"
    />
  </div>
</div>
`;

exports[`Loader Loader is large 1`] = `
<div>
  <div
    class="loader-container my-loader-class"
    data-testid="test-loader"
  >
    <div
      class="dot large"
    />
    <div
      class="dot large"
    />
    <div
      class="dot large"
    />
  </div>
</div>
`;

exports[`Loader Loader is medium 1`] = `
<div>
  <div
    class="loader-container my-loader-class"
    data-testid="test-loader"
  >
    <div
      class="dot medium"
    />
    <div
      class="dot medium"
    />
    <div
      class="dot medium"
    />
  </div>
</div>
`;

exports[`Loader Loader is small 1`] = `
<div>
  <div
    class="loader-container my-loader-class"
    data-testid="test-loader"
  >
    <div
      class="dot small"
    />
    <div
      class="dot small"
    />
    <div
      class="dot small"
    />
  </div>
</div>
`;

exports[`Loader Renders without crashing 1`] = `
<div>
  <div
    class="loader-container my-loader-class"
    data-testid="test-loader"
  >
    <div
      class="dot small"
    />
    <div
      class="dot small"
    />
    <div
      class="dot small"
    />
  </div>
</div>
`;
